src/test/daml/Daml/Finance/Settlement/Test/Transfer.daml

```diff
@@ -316,7 +326,84 @@ run4 = script do
 
   pure ()
 
+-- Transfer of commercial bank money from Alice to Charlie via Bob (acting as an intermediary/passthrough).
+-- +---------------------------------------------------+
+-- | Accounts                                          |
+-- +---------------------------+-----------------------+
+-- | structure:                | used for:             |
+-- +---------------------------+-----------------------+
+-- |           Bank            |                       |
+-- |      /     |    \         | commercial bank money |
+-- | Sender -> CCP -> Receiver |                       |
+-- +---------------------------+-----------------------+
+run5 : Script ()
+run5 = script do
+  TestParties{..} <- setupParties
+
+  -- Account and holding factory
+  let fp = [("FactoryProvider", singleton publicParty)]
+  accountFactoryCid <- toInterfaceContractId <$> Account.createFactory provider fp
+  holdingFactoryCid <- toInterfaceContractId <$> submitMulti [publicParty] [] do
+    createCmd Fungible.Factory with provider = publicParty; observers = M.fromList fp
+
+  -- Create accounts
+  [senderAccount, ccpAccount, receiverAccount] <- mapA (Account.createAccount "Cash Account" [publicParty] accountFactoryCid holdingFactoryCid [] Account.Owner bank) [sender, ccp, receiver]
+
+  -- Distribute asset
+  now <- getTime
+  cashInstrument <- Instrument.originate cb cb "USD" "United States Dollar" [] now
+  transferableCid <- coerceContractId <$> Account.credit [publicParty] cashInstrument 200_000.0 senderAccount
+
+  -- Create settlement factory
+  settlementFactoryCid <- toInterfaceContractId @Factory.I <$> submitMulti [bank] [] do createCmd Factory with provider = bank; observers = empty
+
+  -- Instruct transfer
+  let
+    step1 = Step with sender; receiver = ccp; quantity = Instrument.qty 200_000.0 cashInstrument
+    step2 = Step with sender = ccp; receiver; quantity = Instrument.qty 200_000.0 cashInstrument
+  (batchCid, [instructionCid2, instructionCid1]) <- submitMulti [bank] [] do
```

Comment:
```suggestion
  (batchCid, [instructionCid2, instructionCid1]) <- submit bank do
```

Also below I have the feeling the allocate and approve do not require public party disclosure (nor submitmulti)